### ISSUE TITLE
String formatting uses computer's locale

### ISSUE THREAD

**Issue by kirberich** [databricks/sjsonnet#19]
Hi! 

I just had an interesting bit of confusion where on a colleague's computer sjsonnet's output differed from my machine, which ultimately turned out to be because of std.format using the computer's locale in a circumstance where it shouldn't.

Here's a minimal code example that replicates the behaviour:

```
{
  "test": std.format("%g", 1.5)
}

# ------ Output with a European locale
~/Code ❯❯❯ sjsonnet.jar test.jsonnet
{
   "test": "1,5"
}

# ------ Output with a UK/US locale
~/Code ❯❯❯ sjsonnet.jar test.jsonnet
{
   "test": "1.5"
}
```

Jsonnect's spec specifies that the string formatting should "follow the same rules as Python.", in which case `%g` shouldn't be locale-specific.

I've compared this behaviour to the standard jsonnet compiler as well, which doesn't exhibit this behaviour. It's definitely quite an esoteric edge case 🤣

**Comment by sbarzowski** [comment]
> It's definitely quite an esoteric edge case rofl

I wouldn't say it's esoteric. It seems like half the countries in the world use the comma (https://en.wikipedia.org/wiki/Decimal_separator#Arabic_numerals).  And many developers use their country's standard locale.

**Comment by lihaoyi-databricks** [comment]
I believed this should be fixed in 0.1.5, where I've re-implemented the string-formatting logic myself and no longer rely on locale-specific library code. It should always do `1.5` with a `.` as the decimal point
